### ISSUE TITLE
Cython 3: nogil must be at the end of the function signature line

### ISSUE THREAD

**Issue by bdice** [NVIDIA/cuda-python#44]
I am experimenting with Cython 3.0.0 beta 1. It raises the following warning hundreds of times when `cimport`-ing cuda-python.

```
warning: /home/bdice/mambaforge/envs/cudf_cython_beta/lib/python3.10/site-packages/cuda/ccudart.pxd:1123:154: The keyword 'nogil' should appear at the end of the function signature line. Placing it before 'except' or 'noexcept' will be disallowed in a future version of Cython.
```

I think this expects `nogil except ?cudaErrorCallRequiresNewerDriver` to be replaced by `except ?cudaErrorCallRequiresNewerDriver nogil`. If we could fix this, it would be a massive improvement to the compiler diagnostics that are shown during builds, which are otherwise overwhelmed by the sheer volume of this message.

**Comment by vzhurba01** [comment]
Resolved in [CUDA Python 12.2.0](https://github.com/NVIDIA/cuda-python/tree/v12.2.0). Closing.

**Comment by bdice** [comment]
@vzhurba01 Thank you!
